tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,206 @@
+"""
```

Comment:
Workflow:

1. Deploy cluster where bug exists. 
2. Try reproduce the issue. 
3. Write data into the cluster. create few pools. this would ensure drain takes some time to complete.
2.0. Select 1 host for drain. (host-1). perform mgr failover. We should see the issue now.
2.1. Check the OSD logs and check for crashes here.
2.2. make sure that ceph orch commands are stuck.
2.3. In finally, apply the WA.
4. Proceed to upgrade the cluster.
5. Now, post upgrade, Start the removal of the same host (host-1), which failed earlier.
6. Now with upgrade, the host removal should work as expected, since the bug is fixed.
7. Add back the host (host-1), to complete the LC of removal and addition of host.

Reply:
Code is added with the steps mentioned above.